src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscFilingControllerImpl.java

```diff
@@ -155,13 +156,16 @@ private Links saveFilingWithLinks(final PscIndividualFiling entity, final String
 
     private Links buildLinks(final PscIndividualFiling savedFiling, final HttpServletRequest request) {
         final var objectId = new ObjectId(Objects.requireNonNull(savedFiling.getId()));
-        final var uriBuilder = UriComponentsBuilder.fromUriString(request.getRequestURI())
-                .pathSegment(objectId.toHexString());
-        final var selfUri = uriBuilder.build().toUri();
-        final var privateUriBuilder =
-                UriComponentsBuilder.fromUriString(PREFIX_PRIVATE + "/" + request.getRequestURI())
-                        .pathSegment(objectId.toHexString());
-        final var validateUri = privateUriBuilder.pathSegment(VALIDATION_STATUS)
+        final var selfUri = UriComponentsBuilder
+                .fromUriString(request.getRequestURI())
+                .pathSegment(objectId.toHexString())
+                .build().toUri();
+
+        final var validateUri = UriComponentsBuilder
+                .fromUriString(PREFIX_PRIVATE + "/" + request.getRequestURI()
+                .replace(StringUtils.join("/", PscTypeConstants.INDIVIDUAL.getValue()), ""))
```

Comment:
How to handle other PSC types later? Maybe build up URI from segments rather than replace request URI

Reply:
Yes, I was conscious that it won't handle other PSC types, so as discussed there's more refactoring than just this when we do other PSC types, and I've added TODO comment.

Reply:
Agreed +1